### PR TITLE
Add permissions with roles for spring.security users.

### PR DESCRIPTION
**Pull request by herdt-michael**
Add assigned permissions with roles spring.security users, instead of assigning all permissions each time without roles.

Signed-off-by: Michael Herdt <michael.herdt2@bosch-si.com>

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java [hunk 1]
```diff
@@ -93,13 +93,28 @@ UserDetailsService userDetailsService() {
         if (userPrincipals.isEmpty()) {
             final String name = securityProperties.getUser().getName();
             final String password = securityProperties.getUser().getPassword();
-            userPrincipals.add(new UserPrincipal(name, password, name, name, name, null, DEFAULT_TENANT,
-                    PermissionUtils.createAllAuthorityList()));
+            List<GrantedAuthority> authorityList;
+            if (securityProperties.getUser().getRoles().isEmpty()) {
```
Comment (schabdo) [inline hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java]: Better extract `securityProperties.getUser().getRoles()` to a local variable and put it next to the username and password variable. Than your if-statement looks cleaner:

```java
if(roles.isEmpty()){
   authorityList = PermissionUtils.createAllAuthorityList();
} else {
   authorityList = createAuthoritiesFrom(roles);
} 
```

File: hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java [hunk 2]
```diff
@@ -93,13 +93,28 @@ UserDetailsService userDetailsService() {
...
+            }
+            userPrincipals
+                    .add(new UserPrincipal(name, password, name, name, name, null, DEFAULT_TENANT, authorityList));
         }
 
         return new FixedInMemoryUserPrincipalUserDetailsService(userPrincipals);
     }
 
+    private static List<GrantedAuthority> createAuthoritiesFromList(final List<String> userAuthorities) {
+        List<GrantedAuthority> grantedAuthorityList = new ArrayList<>(userAuthorities.size());
```
Comment (schabdo) [inline hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java]: This looks almost the same as above (lines 78-82) can't you reuse same method there somehow as well?

File: hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java [hunk 3]
```diff
@@ -93,13 +90,29 @@ UserDetailsService userDetailsService() {
...
             final String password = securityProperties.getUser().getPassword();
-            userPrincipals.add(new UserPrincipal(name, password, name, name, name, null, DEFAULT_TENANT,
-                    PermissionUtils.createAllAuthorityList()));
+            final List<String> roles = securityProperties.getUser().getRoles();
+            List<GrantedAuthority> authorityList;
+            if (roles.isEmpty()) {
+                authorityList = PermissionUtils.createAllAuthorityList();
+            } else {
+                authorityList = createAuthoritiesFromList(roles);
+            }
```
Comment (a-sayyed) [inline hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java]: for unary conditions like this, I prefer to use the conditional operator:
```
List<GrantedAuthority> authorityList = roles.isEmpty() ? PermissionUtils.createAllAuthorityList() : createAuthoritiesFromList(roles);
```
But that's of course a coding style and in no way a requirement ;)

File: hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java (hunk 2 above)
Comment (a-sayyed) [reply hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java]: Isn't it the case that 'previous' lines 78-82 were refactored to be this method?

Comment (schabdo) [reply hawkbit-autoconfigure/src/main/java/org/eclipse/hawkbit/autoconfigure/security/InMemoryUserManagementAutoConfiguration.java]: @herdt-michael was really fast by doing this

**Pull request merged**